editor/src/components/canvas/canvas-strategies/strategies/reparent-metastrategy.tsx

```diff
@@ -106,6 +121,75 @@ export function getApplicableReparentFactories(
   return factories
 }
 
+function getStartingTargetParentsToFilterOutInner(
+  canvasState: InteractionCanvasState,
+  interactionSession: InteractionSession,
+): ReparentTargetsToFilter {
+  const interactionData = interactionSession.interactionData
+  if (interactionData.type !== 'DRAG') {
+    throw new Error(
+      `getStartingTargetParentsToFilterOut should only be called from a DRAG type interaction, not a ${interactionData.type} type interaction`,
+    )
+  }
+
+  const pointOnCanvas = interactionData.originalDragStart
+  const allowSmallerParent = interactionData.modifiers.cmd
+    ? 'allow-smaller-parent'
+    : 'disallow-smaller-parent'
+
+  const reparentSubjects = reparentSubjectsForInteractionTarget(canvasState.interactionTarget)
+
+  const strictBoundsResult = getReparentTargetUnified(
+    reparentSubjects,
+    pointOnCanvas,
+    interactionData.modifiers.cmd,
+    canvasState,
+    canvasState.startingMetadata,
+    canvasState.startingAllElementProps,
+    'use-strict-bounds',
+    allowSmallerParent,
+  )
+
+  const missingBoundsResult = getReparentTargetUnified(
+    reparentSubjects,
+    pointOnCanvas,
+    interactionData.modifiers.cmd,
+    canvasState,
+    canvasState.startingMetadata,
+    canvasState.startingAllElementProps,
+    'allow-missing-bounds',
+    allowSmallerParent,
+  )
+
+  return reparentTargetsToFilter(strictBoundsResult, missingBoundsResult)
+}
+
+function isCanvasState(
+  v: InteractionCanvasState | InteractionSession,
+): v is InteractionCanvasState {
+  return (v as InteractionCanvasState).startingMetadata != null
+}
+
+const getStartingTargetParentsToFilterOut = memoize(getStartingTargetParentsToFilterOutInner, {
+  maxSize: 5,
+  equals: (
+    l: InteractionCanvasState | InteractionSession,
+    r: InteractionCanvasState | InteractionSession,
+  ) => {
+    // We only need to re-calculate the targets to filter if the interaction targets or starting metadata have changed
+    if (isCanvasState(l) && isCanvasState(r)) {
+      // FIXME What about insertion?
```

Comment:
I've realised that insertion shouldn't be filtering any initial targets. Will fix that now.

Reply:
Fixed